File: LeetCode/Python Solutions/Minimize Maximum Pair Sum in Array/minimize.py
class Solution:
    def minPairSum(self, nums: List[int]) -> int:
        nums.sort()  # Sort the array
        n = len(nums)
        maxPairSum = 0

        # Pair the smallest element with the largest, and so on.
        for i in range(n // 2):
            maxPairSum = max(maxPairSum, nums[i] + nums[n - 1 - i])

        return maxPairSum

File: LeetCode/Java Solutions/Minimize Maximum Pair Sum in Array/minimize.java
import java.util.Arrays;

class Solution {
    public int minPairSum(int[] nums) {
        Arrays.sort(nums); // Sort the array
        int n = nums.length;
        int maxPairSum = 0;

        // Pair the smallest element with the largest, and so on
        for (int i = 0; i < n / 2; i++) {
            maxPairSum = Math.max(maxPairSum, nums[i] + nums[n - 1 - i]);
        }

        return maxPairSum;
    }
}
